kernel/environ.ml

```diff
@@ -400,15 +400,19 @@ let add_universes ~lbound ~strict ctx g =
 let push_context ?(strict=false) ctx env =
   map_universes (add_universes ~lbound:(universes_lbound env) ~strict ctx) env
 
-let add_universes_set ~lbound ~strict ctx g =
+let add_universes_set ~enforce_fresh ~lbound ~strict ctx g =
   let g = Univ.LSet.fold
             (* Be lenient, module typing reintroduces universes and constraints due to includes *)
-            (fun v g -> try UGraph.add_universe ~lbound ~strict v g with UGraph.AlreadyDeclared -> g)
+            (fun l g -> try UGraph.add_universe ~lbound ~strict l g
+              with UGraph.AlreadyDeclared ->
+                if enforce_fresh then
+                  CErrors.anomaly Pp.(str "Universe " ++ Univ.Level.pr l ++ str" is already declared.")
```

Comment:
I  think we should just let AlreadyDeclared pass through when `enforce_fresh`, there's no point in having the anomaly.
Then you can write it as `try ... with UGraph.AlreadyDeclared when not enforce_fresh -> g`
This matches the behaviour of push_context